src/Chapter4.hs

```diff
@@ -629,7 +658,8 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+-- andM mb mb' = mb >>= (\b -> (b &&) <$> mb')
+andM mb mb' = mb >>= \b -> if b then mb' else pure False -- coulnd't decide which one I like more. This is the lazy one if I understand correctly
```

Comment:
Yes, this is the more correct one, as it is lazy, and we expected the lazy solution in here 🙂 
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour:slightly_smiling_face: 

In the first version of yours, you don't actually use the power of Monad and `>>=` in there could be easily replaced, saving the same further implementation. If you don't use the `>>=` operator, then the `Monad` constraint is too strong, and it is enough to specify only `Applicative`. However, with `Applicative` you are not able to implement a lazier version of `andM`.

Reply:
Yes, that's a good point. It's interesting to see that it is possible to implement it using `Applicative` exclusively, but only in a non-lazy version